l3kernel/l3pdf.dtx

```diff
@@ -314,38 +360,59 @@
 %
 % \subsection{Objects}
 %
-% \begin{macro}{\pdf_object_new:n}
-% \begin{macro}
-%   {\pdf_object_write:nnn, \pdf_object_write:nne, \pdf_object_write:nnx}
-% \begin{macro}{\pdf_object_ref:n}
+% \begin{variable}{\g_@@_backend_object_int}
+%   For returning object numbers.
+%    \begin{macrocode}
+\int_new:N \g_@@_backend_object_int
+%    \end{macrocode}
+% \end{variable}
+%
+% \begin{macro}{\pdf_object_new:n, \@@_object_new:n, \@@_object_new:e}
 % \begin{macro}
 %   {
-%     \pdf_object_unnamed_write:nn, \pdf_object_unnamed_write:ne,
-%     \pdf_object_unnamed_write:nx
+%     \pdf_object_write:nnn, \pdf_object_write:nne, \pdf_object_write:nnx
+%     \@@_object_write:nnn, \@@_object_write:enn
 %   }
-% \begin{macro}{\pdf_object_ref_last:}
-% \begin{macro}[pTF]{\pdf_object_if_exist:n}
+% \begin{macro}{\pdf_object_ref:n, \@@_object_ref:n, \@@_object_ref:e}
 %   Simple to do: all objects create a constant |int| so it is not a
 %   backend-specific name.
 %    \begin{macrocode}
 \cs_new_protected:Npn \pdf_object_new:n #1
+  { \@@_object_new:e {#1} }
+\cs_new_protected:Npn \@@_object_new:n #1
   {
-    \exp_args:Ne \@@_backend_object_new:n {#1}
+    \@@_backend_object_new:
+    \int_const:cn
+      { c_@@_object_ #1 _int }
+      { \g_@@_backend_object_int }
   }
+\cs_generate_variant:Nn \@@_object_new:n { e }
```

Comment:
Seems `\__pdf_object_new:n` can be merged to `\pdf_object_new:n`.

The `#1` is always fully expanded by the `c`-variant of `\int_const:cn` so `e`-type expansion is not necessary.

Reply:
Agreed - changed